sky/provision/kubernetes/utils.py

```diff
@@ -3179,7 +3183,27 @@ def filter_pods(namespace: str,
     pods = [
         pod for pod in pod_list.items if pod.metadata.deletion_timestamp is None
     ]
-    return {pod.metadata.name: pod for pod in pods}
+
+    # Sort pods by name, with workers sorted by their numeric suffix.
+    # This ensures consistent ordering (e.g., cluster-head, cluster-worker1,
+    # worker2, worker3, ...) even when Kubernetes API returns them in
+    # arbitrary order. This works even if there were somehow pod names other
+    # than head/worker ones, but that may be overkill.
+    def get_pod_sort_key(pod: V1Pod) -> Tuple[int, Union[int, str]]:
```

Comment:
Hmm, this `Union[int, str]` seems a bit concerning, i.e. what if some pods returns a str but some returns a int? We will get the following error during sorting?
```python
>>> a = (0, 1)
>>> b = (0, 'my-worker')
>>> a > b
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '>' not supported between instances of 'int' and 'str'
```